packages/flutter/test/widgets/media_query_test.dart

```diff
@@ -24,31 +24,58 @@ void main() {
     expect(exception, isNotNull);
     expect(exception ,isFlutterError);
     final FlutterError error = exception as FlutterError;
-    expect(error.diagnostics.length, 3);
-    expect(error.diagnostics.last, isA<DiagnosticsProperty<Element>>());
+    expect(error.diagnostics.length, 5);
+    expect(error.diagnostics.last, isA<ErrorHint>());
     expect(
       error.toStringDeep(),
       equalsIgnoringHashCodes(
         'FlutterError\n'
-        '   MediaQuery.of() called with a context that does not contain a\n'
-        '   MediaQuery.\n'
-        '   No MediaQuery ancestor could be found starting from the context\n'
-        '   that was passed to MediaQuery.of(). This can happen because you\n'
-        '   do not have a WidgetsApp or MaterialApp widget (those widgets\n'
-        '   introduce a MediaQuery), or it can happen if the context you use\n'
-        '   comes from a widget above those widgets.\n'
-        '   The context used was:\n'
-        '     Builder\n',
+        '   No MediaQuery widget ancestor found.\n'
+        '   Builder widgets require a MediaQuery widget ancestor.\n'
+        '   The specific widget that could not find a MediaQuery ancestor\n'
+        '   was:\n'
+        '     Builder\n'
+        '   The ownership chain for the affected widget is: "Builder ←\n'
+        '     [root]"\n'
+        '      No MediaQuery ancestor could be found starting from the\n'
+        '   context\n'
+        '      that was passed to MediaQuery.of(). This can happen because\n'
+        '   you\n'
+        '      have not added a WidgetsApp, CupertinoApp, or MaterialApp\n'
+        '   widget\n'
+        '      (those widgets introduce a MediaQuery), or it can happen if\n'
+        '   the\n'
+        '      context you use comes from a widget above those widgets.\n'
```

Comment:
we definitely shouldn't expect that output, it looks horrible.

Reply:
LOL, good point.  I think part of that is because the testing environment uses a smaller terminal width.  Probably what we should be doing is not putting in any `\n`'s in the message and letting it wrap algorithmically.